Feature/kloubak pr

m3d opened the pull request:
This is driver for new articulated robot "Kloubak" with 4 BLDC motors. By accident I forgot "config/kloubak-subt-camera.json". I would squash all related commits. (I will also fixup missing line in `osgar/drivers/__init__.py` which I accidentally removed.

zwn commented on osgar/drivers/kloubak.py:
```diff
@@ -0,0 +1,283 @@
+  (https://github.com/tf-czu/kloubak)
+"""
+
+import ctypes
```
This import is not used.

zwn commented on osgar/drivers/kloubak.py:
```diff
@@ -0,0 +1,283 @@
+import ctypes
+import struct
+import math
+from datetime import timedelta
```
This import is not used.

zwn commented on osgar/drivers/kloubak.py:
```diff
@@ -0,0 +1,283 @@
+
+    def process_packet(self, packet, verbose=False):
+        if len(packet) >= 2:
+            msg_id = ((packet[0]) << 3) | (((packet[1]) >> 5) & 0x1f)
```
We have this exact same line in eduro.py and spider.py. Maybe it is time for refactoring? Or would you rather keep copy pasting?

m3d replied:
good point! I would like to "solve" it with changing the structure how CAN messages are sent. Instead of raw couple of bytes it will be (msg_id, data, flags), so this function would no longer be necessary. This CAN refactoring I would do after working pcan on linux (APU) in separate PR - so yes, this should away (later).

zwn replied:
👍

zwn commented on osgar/drivers/kloubak.py:
```diff
@@ -0,0 +1,283 @@
+                        else:
+                            self.publish('can', CAN_packet(0x12, stop))  # left front
+                    else:
+                        self.publish('can', CAN_packet(0x12, fwd))  # left front"""
```
What is the plan with the commented out portions inhere?

m3d replied:
they are obsolete. They were used for current control and we will not return to this.

zwn replied:
So it might be ok to delete them?

m3d replied:
yes - this PR is collection of "feature/kloubak" branch, so I would like to fix it there and then cherrypick this fixes. I will add them probably today, thanks.

m3d commented:
OK, I rebased it so that we do not have conflict with master any more, and added couple of commits with proportional steering. With squash I would wait for test confirmation.
